### PR TITLE
[TOOLS-4572] improve charactor column size control function

### PR DESCRIPTION
**Pull request by Kang-dot**
http://jira.cubrid.org/browse/TOOLS-4572

- Purpose
cubrid version has been updated, it is no longer necessary to adjust the parameter type size in the latest version, so add conditions to the dialog to improve the function.

- Implementation
N/A

- Remarks
N/A

### CODE REVIEW

File: com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/message/Messages.properties [hunk 1]
```diff
@@ -484,7 +484,7 @@ msgCanceled=Canceled
...
 msgCfmDeleteSQL=Do you want to delete selected SQL?
 msgCfmErrorSave=Error occurred when saving configuration and some configuration may be lost.\r\nDo you want to leave this page?
-msgCheckCharset=** Notice **\n\nTo avoid miss any data unexpectedly, please confirm charactor columns' size before migration.\n\nBefore CUBRID 9.3, the length definition of a charactor column means how many bytes can be stored into the column, for example\: char(10) can store 10 single byte charactors or 5 double bytes charactors.\nSince CUBRID 9.3, the length means how many charactors can be stored into the column.\n\nDo you want to go to set up the columns length now?
+msgCheckCharset=** Notice **\n\nTo avoid miss any data unexpectedly, please confirm charactor columns' size before migration.\n\Less than CUBRID 9.3, the length definition of a charactor column means how many bytes can be stored into the column, for example\: char(10) can store 10 single byte charactors or 5 double bytes charactors.\nSince CUBRID 9.3, the length means how many charactors can be stored into the column.\n\nDo you want to go to set up the columns length now?
```
Comment (Srltas) [inline com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/message/Messages.properties]: The line break string 'n' is missing, so there is no line space.

File: com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java [hunk 2]
```diff
@@ -889,8 +889,17 @@ protected VerifyResultMessages createVerifyResult(
 
     /** @return true if user should change the target character type's size to avoid data lost. */
     public boolean doesNeedToChangeCharacterTypeSize() {
-        return DatabaseType.MYSQL.equals(config.getSourceDBType())
-                || DatabaseType.ORACLE.equals(config.getSourceDBType());
+        boolean isEffectedByCharacterTypeSize;
+
+        if (config.targetIsDBDump() || config.targetIsCSV()) {
```
Comment (Srltas) [inline com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: Changing it to `config.targetIsOnline()` is clearer.
If you use the code you wrote, you should also add SQL and XLS.

File: com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java [hunk 3]
```diff
@@ -889,8 +889,17 @@ protected VerifyResultMessages createVerifyResult(
...
+            isEffectedByCharacterTypeSize = !config.isAddUserSchema();
+        } else {
+            isEffectedByCharacterTypeSize =
+                    Integer.parseInt(config.getTargetDBVersion()) < 93 ? true : false;
```
Comment (Srltas) [inline com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: There is no problem with the code working without `? true : false`.

File: com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/message/Messages.properties (hunk 1 above)
Comment (Kang-dot) [reply com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/message/Messages.properties]: i'll add missing charactor in next commit

File: com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java (hunk 2 above)
Comment (Kang-dot) [reply com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: change if condition statement. it cover all 'target file' migration

File: com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java (hunk 3 above)
Comment (Kang-dot) [reply com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: i'll remove unnecessary code in next commit

File: com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java [hunk 4]
```diff
@@ -889,8 +889,16 @@ protected VerifyResultMessages createVerifyResult(
...
     /** @return true if user should change the target character type's size to avoid data lost. */
     public boolean doesNeedToChangeCharacterTypeSize() {
-        return DatabaseType.MYSQL.equals(config.getSourceDBType())
-                || DatabaseType.ORACLE.equals(config.getSourceDBType());
+        boolean isEffectedByCharacterTypeSize;
+
+        if (config.targetIsFile()) {
+            isEffectedByCharacterTypeSize = !config.isAddUserSchema();
+        } else {
+            isEffectedByCharacterTypeSize = Integer.parseInt(config.getTargetDBVersion()) < 93;
```
Comment (Srltas) [inline com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: How about making the integer '93' a constant and using it?
ex) `CUBRID_9_3_UNDER`

Comment (Kang-dot) [reply com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: i think that is good suggestion and i agree that. but i think if we make '93' to static string variable, either '112' will apply that change. and this change need working in another issue.

Comment (Srltas) [reply com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: I don't understand that 112 is affected by the correction.
In the view of other developers, the integer '93' is thought to be misleading, so the opinion is to explain it with the variable name.

Comment (Kang-dot) [reply com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: i'm checking code now, some files use '112' as it is, and other some file use static variable. they are not consistent. 
so how about this? in this issue, make '93' to constant variable. and the '112' values which is not variable will change to constant variable in other issue.

Comment (Srltas) [reply com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/utils/MigrationCfgUtils.java]: In the current issue, we generate a constant for '93', and we solve the constant problem of '112' by creating another issue.

**Review by Srltas — APPROVED** [review]

**Review by kisoo-han — APPROVED** [review]

**Pull request merged**